src/compiler/checker.ts

```diff
@@ -7983,6 +8188,60 @@ namespace ts {
             return links.resolvedType;
         }
 
+        function getWidenedTypeOfUniqueESSymbolType(type: Type): Type {
+            return type.flags & TypeFlags.UniqueESSymbol ? esSymbolType :
+                type.flags & TypeFlags.Union ? getUnionType(sameMap((<UnionType>type).types, getWidenedTypeOfUniqueESSymbolType)) :
+                type.flags & TypeFlags.Intersection ? getIntersectionType(sameMap((<IntersectionType>type).types, getWidenedTypeOfUniqueESSymbolType)) :
+                type;
+        }
+
+        function createUniqueESSymbolType(symbol: Symbol) {
+            const type = <UniqueESSymbolType>createType(TypeFlags.UniqueESSymbol);
+            type.symbol = symbol;
+            return type;
+        }
+
+        function isReferenceToValidDeclarationForUniqueESSymbol(symbol: Symbol) {
+            if (symbol.valueDeclaration) {
+                switch (symbol.valueDeclaration.kind) {
+                    case SyntaxKind.VariableDeclaration:
+                        return getNameOfDeclaration(symbol.valueDeclaration).kind === SyntaxKind.Identifier
+                            && symbol.valueDeclaration.parent.kind === SyntaxKind.VariableDeclarationList

```

Comment:
these two lines are equivalent to the new utility `isVariableDeclarationInVariableStatement`.